control-plane/catalog/to-consul/resource.go

```diff
@@ -641,6 +667,24 @@ func (t *ServiceResource) generateRegistrations(key string) {
 	}
 }
 
+// Sets the passing service weight
+func (t *ServiceResource) setServiceWeight(weight string, appsCount int, r consulapi.CatalogRegistration) error {
+	// error validation if the input param is a number
+	weightI, err := strconv.Atoi(weight)
+	if err != nil {
+		return err
+	} else if weightI > 1 {
```

Comment:
It is always good to keep the happy path aligned towards the left which roughly translates to avoiding `else` and nested if blocks. This improves readability a lot. You could refactor this block something like

```go
weightI, err := strconv.Atoi(weight)
if err != nil {
     return err
}

if weightI <=1 {
   return nil
}

var perAppWeight = weightI / appsCount
if perAppWeight < 1 {
   perAppWeight = 1
}
r.Service.Weights = consulapi.AgentWeights{
	Passing: perAppWeight,
}

return nil
```

to make it more readable.